x-pack/plugins/apm/public/components/app/service_inventory/ServiceList/service_list.test.tsx

```diff
@@ -37,45 +37,20 @@ describe('ServiceList', () => {
   it('renders with data', () => {
     expect(() =>
       renderWithTheme(
-        <ServiceList items={props.items as ServiceListAPIResponse['items']} />,
+        <ServiceList
+          items={(props.items as unknown) as ServiceListAPIResponse['items']}
```

Comment:
why do we now need to cast to unknown?

Reply:
@dgieselaar to remove the `unknown` here I had to manually define the return type of the API https://github.com/elastic/kibana/pull/86434/commits/dfe8666ffaeda7c4195326ce10872bfd1a338af6#diff-78dffa74f895585e8eeb6efef9323a17f0255b056f03c5ace90e4d38d6c2d5b3R26-R35.

This was needed because even though the `healthStatus` is an optional property, the inferred type defined it as required.

Reply:
Hmm, I'm still not sure why it's necessary. I'll have a look.

Reply:
I think we're better off by changing props.json to a ts file and explicitly typing it as `type ServiceListAPIResponse = APIReturnType<'GET /api/apm/services'>`. It will create an error about healthStatus, that means the type that `joinByKey` returns is incorrect. That should probably be addressed, can you look into it? It's a complicated type so I wouldn't spend too much time on it. But the idea is that only properties that are defined in all types should be required. All others should be optional because we cannot guarantee them.

Reply:
I got what was the problem, event though TS was complaining about the `healthStatus` field, the field that was actually missing was the `transactionType`.

Reply:
But I converted the `props.json` to a TS.